vault/keysource.go

```diff
@@ -0,0 +1,272 @@
+package vault
```

Comment:
I believe it would be worth renaming this package to `hcvault` to bring it into alignment with the CLI and from our learned lesson on having `kms` and `gcpkms` packages.